File: tools/drives/upstream/match-text-one-word.js
// The labels marked on a one-word query, for the shot the port is read against.
//
//   pnpm qa --start --path /widgets/match-text/ --framework react \
//     --drive .../match-text-one-word.js --shot /tmp/ref/match-text-one-word.png
// The docs chrome carries a search box of its own, so the query box is taken from the demo pane.
const pane = $('[data-pane="react"]') ?? document;
const input = $('[data-slot="input"]', pane);
if (!input) return { verdict: 'FAIL the page has no query box' };
const setter = Object.getOwnPropertyDescriptor(Object.getPrototypeOf(input), 'value').set;
setter.call(input, 'mo');
input.dispatchEvent(new Event('input', { bubbles: true }));
await wait(400);
const labels = $$('[data-slot="match-text"]', pane);
const marked = labels.flatMap((el) =>
  [...el.children].filter((run) => Number(getComputedStyle(run).fontWeight) > Number(getComputedStyle(el).fontWeight)).map((run) => run.textContent),
);
input.scrollIntoView({ block: 'start' });
await wait(300);
return {
  verdict: marked.length > 0 ? 'PASS a one-word query marks that word wherever it occurs' : 'FAIL nothing was marked',
  marked,
};

File: tools/drives/match-text-one-word.py
"""The state matrix: the labels marked on a one-word query.

    .venv/bin/python tools/qa.py --page match-text \
      --drive tools/drives/match-text-one-word.py --shot /tmp/match-text-one-word.png

The twin of `tools/drives/upstream/match-text-one-word.js`.
"""
from __future__ import annotations

from sg_widgets_qt.widgets.match_text import MatchText

#: One word, which the endpoint matches wherever it occurs (053_text_search_matching).
WORD = "mo"


def drive(page, wait, find, prefs) -> dict:  # noqa: ARG001
    field = find("match-query")
    if field is None:
        return {"verdict": "FAIL the page has no query box"}
    field.set_text(WORD)
    wait(300)
    marked = [run.text for one in find(MatchText, all=True) for run in one.runs if run.match]
    return {
        "verdict": (
            "PASS a one-word query marks that word wherever it occurs"
            if marked and all(run.lower() == WORD for run in marked)
            else f"FAIL a one-word query marked {marked}"
        ),
        "seen": {"query": WORD, "marked": marked},
    }
